src/NexusMods.App.UI/Pages/Downloads/DownloadsPageViewModel.cs

```diff
@@ -11,28 +18,211 @@ namespace NexusMods.App.UI.Pages.Downloads;
 
 public class DownloadsPageViewModel : APageViewModel<IDownloadsPageViewModel>, IDownloadsPageViewModel
 {
-    private readonly IDownloadsService _downloadsService;
-
     [Reactive] public int SelectionCount { get; private set; } = 0;
     
     [Reactive] public bool IsEmptyStateActive { get; set; } = true;
+    
+    public string HeaderTitle { get; private set; } = string.Empty;
+    public string HeaderDescription { get; private set; } = string.Empty;
 
-    public ReactiveCommand<Unit> PauseAllCommand { get; }
-    public ReactiveCommand<Unit> ResumeAllCommand { get; }
-    public ReactiveCommand<Unit> CancelSelectedCommand { get; }
+    public DownloadsTreeDataGridAdapter Adapter { get; }
+    public ReactiveCommand<Unit> PauseAllCommand { get; private set; } = null!;
+    public ReactiveCommand<Unit> ResumeAllCommand { get; private set; } = null!;
+    public ReactiveCommand<Unit> PauseSelectedCommand { get; private set; } = null!;
+    public ReactiveCommand<Unit> ResumeSelectedCommand { get; private set; } = null!;
+    public ReactiveCommand<Unit> CancelSelectedCommand { get; private set; } = null!;
+    public Observable<bool> SelectionHasRunningItems { get; private set; } = null!;
+    public Observable<bool> SelectionHasPausedItems { get; private set; } = null!;
+    public Observable<bool> SelectionHasActiveItems { get; private set; } = null!;
+    public Observable<bool> HasRunningItems { get; private set; } = null!;
+    public Observable<bool> HasPausedItems { get; private set; } = null!;
```

Comment:
I'd rather create an empty command and assign to that than `null!` which is a bit icky

Reply:
These are assigned at activation; and activation is part of bringing up the UI, so there's no way to access it before it's needed.

If anything, I'd consider this preferable. Access before it's initialised results in a clear error; rather than a silently discarded one or unexpected behaviour.